packages/ui-menu/src/Menu/index.tsx

```diff
@@ -470,9 +470,9 @@ class Menu extends Component<MenuProps> {
           }
         }}
         renderTrigger={safeCloneElement(trigger as ReactElement, {
-          ref: (el: ReactElement) => {
+          ref: (el: React.ReactInstance & { ref?: Element }) => {
             this._trigger = el
-            this.ref = el
+            this.ref = el.ref || (el as Element)
```

Comment:
I refactored the ref-s as well, because (I think) these are ReactInstance type (Component<any> | Element). Also added a bit of extra logic here, because if the ref is a Component, we need its `.ref`, not the component itself.